src/chexus/validators.py

```diff
@@ -154,6 +154,22 @@ def validate(self, node: Dataset | Group) -> Violation | None:
             return Violation(node.name)
 
 
+class float_dataset_units_check(float_dataset_units_missing):
+    def __init__(self) -> None:
+        super(float_dataset_units_missing, self).__init__(
+            "float_dataset_units_check",
+            "Float dataset should have units parasable by scipp",
+        )
+
+    def validate(self, node: Dataset | Group) -> Violation | None:
+        import scipp as sc
+
+        try:
+            sc.Unit(node.attrs['units'])
```

Comment:
If this gets relaxed to allow non-float datasets, you need to skip the check if the dataset has no unit. (Could be done in to `applies_to`)